src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -49,7 +49,6 @@ private sealed partial class TagSource
         /// </summary>
         private const int CoalesceDifferenceCount = 10;
 
-        private static readonly ObjectPool<SegmentedList<TagSpan<TTag>>> s_tagSpanListPool = new(() => new(), trimOnFree: false);
```

Comment:
switched to an existing pool we already have for SegmentedLists (which already has the "do not trim on free" concept).